apps/api/src/app/events/services/mail-service/handlers/msgraph.handler.ts

```diff
@@ -0,0 +1,23 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { ChannelTypeEnum } from '@novu/shared';
+import { MSGraphAPIProvider } from '@novu/msgraph';
+import { BaseHandler } from './base.handler';
+import { ICredentials } from '@novu/dal';
+
+export class MSGraphHandler extends BaseHandler {
+  constructor() {
+    super('msgraph', ChannelTypeEnum.EMAIL);
+  }
+  buildProvider(credentials: ICredentials, from?: string) {
+    // null-assertion is disabled because we want to expose the error from the MS Graph API
+    this.provider = new MSGraphAPIProvider({
+      clientId: credentials.clientId!,
+      authority: credentials.applicationId!,
+      secret: credentials.apiKey!,
```

Comment:
```suggestion
      secret: credentials.secretKey!,
```
Api key credential is meant for any public key that can be shared (usually to identify your account in corresponding service).